### ISSUE TITLE
Pylint doesn't know that a numpy method returns a Series if it's passed one.

### ISSUE THREAD

**Issue by x-tabdeveloping** [PyCQA/pylint#7314]
### Bug description

```python
# File: pylint_bug.py
# pylint: disable=missing-docstring
import numpy as np
import pandas as pd

s = pd.Series([1, 2, 3])
log_s = np.log(s)
# Here you can see that the return type is Series
print(type(log_s))  # <class 'pandas.core.series.Series'>
# And therefore it has a .groupby() method
# Pylint on the other hand will say that this line is incorrect
print(log_s.groupby(lambda x: x > 0.5).sum())
# False    0.000000
# True     1.791759
# dtype:   float64

```

### Command used

```shell
pylint pylint_bug.py
```

### Pylint output

```shell
************* Module pylint_bug
pylint_bug.py:12:6: E1101: Instance of 'ndarray' has no 'groupby' member (no-member)

------------------------------------------------------------------
Your code has been rated at 1.67/10
```

### Expected behavior

Pylint should not have informed me about a non-existent error. The code runs just fine, as np.log returns a Series object, which does have a 'groupby' member.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:24:11) 
[GCC 9.4.0]
```

### OS / Environment

Distributor ID: Ubuntu
Description:    Ubuntu 22.04 LTS
Release:        22.04
Codename:       jammy

### Additional dependencies

pandas==1.4.3
numpy==1.22.4

**Comment by mbyrnepr2** [comment]
I'm not sure why but it doesn't happen on the latest main branch of Pylint. Tested on Python 3.9 & 3.10.
I can reproduce the issue using the reported pylint 2.14.5.

**Comment by Pierre-Sassoulas** [comment]
I was thinking that we could add a regression tests at first (to be sure that it's working in all envs/interpreters), but we would need numpy to be installed, right ?

**Comment by mbyrnepr2** [comment]
Numpy-related tests are in Astroid & are skipped if it is not installed; e.g.:https://github.com/PyCQA/astroid/blob/main/tests/unittest_brain_numpy_core_einsumfunc.py#L30

Do you think we should also add Numpy tests in Pylint?

**Comment by Pierre-Sassoulas** [comment]
If we start adding test for every libs it's a slippery slope, so no :)

@x-tabdeveloping could you test by installing the latest dev version using ``pip install -U git+https://github.com/PyCQA/pylint.git@main`` please ?

**Comment by x-tabdeveloping** [comment]
@Pierre-Sassoulas I tested it real quick, and it doesn't happen in the latest version :)

**Comment by Pierre-Sassoulas** [comment]
Good news, thank you for testing ! It's going to be fixed when we release pylint 2.15.0, you can follow the release planning [in the milestone](https://github.com/PyCQA/pylint/milestone/56)
